src/instrument.ts

```diff
@@ -246,3 +250,284 @@ class noopSpan implements Span {
     return this;
   }
 }
+
+interface TraceWrapped extends Function {
+  traceWrapped?: Function;
+}
+
+function performWrap(
+  klass: Function,
+  spanName: string,
+  original: TraceWrapped
+) {
+  /*
+  original would be for example:
+    const [rows] = await database.run('SELECT 1');
+  or
+    database.run('SELECT 1', (err, rows) => {
+        ...
+    });
+  */
+  if (original.traceWrapped) {
+    return original;
+  }
+
+  const traced: TraceWrapped = function (
+    this: any,
+    ...args: any[]
+  ):
+    | void
+    | Promise<unknown>
+    | PartialResultStream
+    | EventEmitter
+    | AsyncIterable<unknown> {
+    const lastArg = args[args.length - 1];
+    const hasCallback = typeof lastArg === 'function';
+    const firstArg = args[0];
+
+    const opts: traceConfig = {opts: this.observabilityOptions || {}};
+
+    if (typeof firstArg === 'string' || (firstArg as SQLStatement)) {
+      opts.sql = firstArg;
```

Comment:
Wouldn't this cause an issue if first arg is not a SQL statement and some other string ?